locksmith/src/operations/pricingOperations.ts

```diff
@@ -16,36 +19,101 @@ export interface Options {
   network: number
 }
 
-export async function getDefiLammaPrice({
+export interface LockPricingProps {
+  amount?: number
+  address?: string
+  lockAddress?: string
+  keysToPurchase?: number
+  network: number
+}
+
+export type PriceResults = Partial<
+  Price & {
+    priceInAmount: number
+  }
+>
+
+/**
+ * Get lock pricing with priority to credit card total if present otherwise return converted price
+ * @returns
+ */
+export async function getUsdLockPricingFromSettingOrConverted({
+  lockAddress,
   network,
-  address,
+  address: currencyContractAddress,
   amount = 1,
-}: Options): Promise<
-  Partial<
-    Price & {
-      priceInAmount: number
+  keysToPurchase = 1,
+}: LockPricingProps): Promise<PriceResults> {
+  const web3Service = new Web3Service(networks)
+  // priority to credit card if present settings is present
+  if (lockAddress) {
+    const { creditCardPrice } = await lockSettingOperations.getSettings({
+      lockAddress,
+      network,
+    })
+
+    if (creditCardPrice) {
+      const creditCardEnabled =
+        await creditCardOperations.getCreditCardEnabledStatus({
+          lockAddress,
+          network,
+          totalPriceInCents: creditCardPrice ?? 0,
+        })
+
+      const lock = await web3Service.getLock(lockAddress, network)
+
+      const symbol =
+        lock?.currencySymbol || networks?.[network]?.nativeCurrency?.symbol
+
+      const price = lock?.keyPrice
+
+      return {
+        price,
+        symbol,
+        creditCardEnabled,
+        decimals: 18,
+        confidence: 1,
+        timestamp: new Date().getTime(),
+        priceInAmount: creditCardPrice * keysToPurchase,
+      }
     }
-  >
-> {
+  }
+
+  // fallback to converted price
+  const result = await getDefiLammaPrice({
+    network,
+    address: currencyContractAddress,
+    amount,
+  })
+
+  return result
+}
+
+export async function getDefiLammaPrice({
+  network,
+  address: currencyContractAddress,
```

Comment:
rename the address to erc20Address perhaps?

Reply:
renamed and replaced also on `openapi`